lib/internal/modules/esm/initialize_import_meta.js

```diff
@@ -71,6 +72,23 @@ function initializeImportMeta(meta, context, loader) {
 
   meta.url = url;
 
+  ObjectDefineProperty(meta, 'require', {
+    __proto__: null,
+    enumerable: true,
+    configurable: true,
+    get() {
+      const { Module: { createRequire } } = require('internal/modules/cjs/loader');
+      this.require = createRequire(url);
+      return this.require;
```

Comment:
We don't have to rely on the setter, which may have been tampered with
```suggestion
      const _require = createRequire(url);
      setOwnProperty(this, 'require', _require);
      return _require;
```